lighthouse/helpers/reports.py

```diff
@@ -254,17 +262,23 @@ def get_all_positive_samples(samples):
 
     return positive_samples_df
 
+
 def add_cherrypicked_column(existing_dataframe):
-    root_sample_ids = existing_dataframe['Root Sample ID'].to_list()
 
-    cherrypicked_samples_df = get_cherrypicked_samples(root_sample_ids)
-    cherrypicked_samples_df['LIMS submission'] = 'Yes'
+    root_sample_ids = existing_dataframe["Root Sample ID"].to_list()
+    plate_barcodes = existing_dataframe["plate_barcode"].unique()
+
+    cherrypicked_samples_df = get_cherrypicked_samples(root_sample_ids, plate_barcodes)
+    cherrypicked_samples_df["LIMS submission"] = "Yes"
 
-    existing_dataframe = existing_dataframe.merge(cherrypicked_samples_df, how="left", on="Root Sample ID")
-    existing_dataframe = existing_dataframe.fillna({'LIMS submission': 'No'})
+    existing_dataframe = existing_dataframe.merge(
+        cherrypicked_samples_df, how="left", on="Root Sample ID"
+    )
+    existing_dataframe = existing_dataframe.fillna({"LIMS submission": "No"})
```

Comment:
Could the overwriting of `existing_dataframe` parameter, then turned variable, be renamed to something more descriptive for both cases? I struggled to follow this bit. Or maybe a method comment too.

Reply:
This was Katy's or Phil's code so didn't really want mess with it. The only changes to this were done by an automated tool.